### PR TITLE
Inputs form with errors

### PR DESCRIPTION
**Pull request by virginieky**

### CODE REVIEW

File: buffet/.storybook/config.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import React from 'react';
 import { addDecorator, configure } from '@storybook/react';
+
+import 'bootstrap/dist/css/bootstrap.css';
```
Comment (Aurelsicoko) [inline buffet/.storybook/config.js]: Do we really need to import the entire Bootstrap library or only the CSS grid?

File: buffet/src/components/Input/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  textarea: Textarea,
+  time: TimePicker,
+  select: Select,
+  email: InputText,
+  password: InputText,
+  search: InputText,
+};
+
+function Input({ onChange, type, ...rest }) {
+  const Compo = inputs[type] || UnknowInput;
```
Comment (Aurelsicoko) [inline buffet/src/components/Input/index.js]: IMHO, we should find another name for this variable like `InputComponent` or `ReactComponent` but the name `Compo` doesn't mean anything to me.

File: buffet/src/styled/TimePicker/TimeList.js [hunk 3]
```diff
@@ -40,20 +41,22 @@ const TimeList = styled.ul`
       opacity: 0;
       z-index: 1;
       cursor: pointer;
-      &:checked + div {
+      background-color: red;
```
Comment (Aurelsicoko) [inline buffet/src/styled/TimePicker/TimeList.js]: Do we really have something with the default `red` value as a background?

File: buffet/src/utils/createYupSchema.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  if (['text', 'textarea', 'email', 'password'].includes(type)) {
+    schema = yup.string(translatedErrors.string);
+  }
+
+  if (type === 'email') {
+    schema = schema.email(translatedErrors.email);
+  }
+
+  if (type === 'number') {
```
Comment (Aurelsicoko) [inline buffet/src/utils/createYupSchema.js]: What about `bigInteger`, `integer`, `float`, `decimal`? See https://strapi.io/documentation/3.x.x/guides/models.html#define-the-attributes

File: buffet/stories/Combined/Form.stories.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  const handleChange = ({ target: { name, value } }) => {
+    dispatch({
+      type: 'ON_CHANGE',
+      key: name,
+      value,
+    });
+  };
+
+  const onSubmit = () => {
+    console.log('HEY');
```
Comment (Aurelsicoko) [inline buffet/stories/Combined/Form.stories.js]: Remove this log?

File: buffet/src/utils/createYupSchema.js (hunk 4 above)
Comment (virginieky) [reply buffet/src/utils/createYupSchema.js]: It concerns the input type (to define validation rules)

File: buffet/src/combined/DateTime/index.js [hunk 6]
```diff
@@ -70,20 +69,22 @@ function DateTime({ name, onChange, value }) {
...
     }
-  }, [value]);
+  }, []);
 
   return (
     <div>
       <DatePicker
         name="date"
         onChange={({ target }) => {
+          console.log('Date changed');
```
Comment (Aurelsicoko) [inline buffet/src/combined/DateTime/index.js]: Log?

File: buffet/stories/Combined/DateTime.stories.js [hunk 7]
```diff
@@ -1,13 +1,12 @@
...
 import { storiesOf } from '@storybook/react';
-import moment from 'moment';
 
 import DateTime from '../../src/combined/DateTime';
 
 function DateTimeStory() {
-  // const [val, setValue] = useState(new Date());
-  const [val, setValue] = useState(moment());
+  const [val, setValue] = useState(new Date());
+  // const [val, setValue] = useState(moment());
```
Comment (Aurelsicoko) [inline buffet/stories/Combined/DateTime.stories.js]: Which line should we keep?

File: buffet/stories/Combined/DateTime.stories.js [hunk 8]
```diff
@@ -18,8 +17,7 @@ function DateTimeStory() {
       {...defaultProps}
       onChange={({ target: { value } }) => {
         setValue(value);
-        console.log(value.get('hour'));
-        console.log(value.get('minute'));
+        console.log(value.format('MMMM Do YYYY, h:mm:ss a'));
```
Comment (Aurelsicoko) [inline buffet/stories/Combined/DateTime.stories.js]: Log?

**Pull request merged**